README.md

```diff
@@ -6,6 +6,221 @@
 [![dependency status](https://deps.rs/repo/github/rust-av/v_frame/status.svg)](https://deps.rs/repo/github/rust-av/v_frame)
 [![codecov](https://codecov.io/github/rust-av/v_frame/branch/main/graph/badge.svg?token=MKT1AZREF0)](https://codecov.io/github/rust-av/v_frame)
 
-Crate providing structs and utilities for handling YUV Frames and Planes.
-Originally used by rav1e, now branched into its own crate for wider use
-throughout Rust AV.
+A Rust library providing efficient data structures and utilities for handling YUV video frames and planes. Originally developed as part of the [rav1e](https://github.com/xiph/rav1e) video encoder, v_frame has been extracted into a standalone crate for broader use across the Rust AV ecosystem.
+
+## Features
+
+- **Type-safe pixel handling**: Generic `Pixel` trait supporting both 8-bit (`u8`) and high bit-depth (`u16`) video
+- **Flexible plane structure**: Efficient memory layout with configurable padding for SIMD operations
+- **Multiple chroma formats**: Support for YUV 4:2:0, 4:2:2, 4:4:4, and monochrome
+- **Builder pattern API**: Safe and ergonomic frame construction with compile-time guarantees
+- **SIMD-friendly alignment**: 64-byte alignment (8-byte on WASM) for optimal performance
+- **WebAssembly support**: Works in both browser (`wasm32-unknown-unknown`) and WASI environments
+- **Zero-copy iterators**: Efficient row-based and pixel-based iteration without allocations
+
+## Installation
+
+Add this to your `Cargo.toml`:
+
+```toml
+[dependencies]
+v_frame = "0.4"
+```
+
+## Quick Start
+
+```rust
+use v_frame::{
+    frame::FrameBuilder,
+    chroma::ChromaSubsampling,
+};
+use std::num::{NonZeroU8, NonZeroUsize};
+
+// Create a 1920x1080 YUV 4:2:0 frame with 8-bit pixels
+let frame = FrameBuilder::new(
+    NonZeroUsize::new(1920).unwrap(),
+    NonZeroUsize::new(1080).unwrap(),
+    ChromaSubsampling::Yuv420,
+    NonZeroU8::new(8).unwrap(),
```

Comment:
Do we want to try to make it const-generic later?

Reply:
Across bit depths? That could be useful. I wonder how much benefit it would give... maybe not a lot directly in v_frame, but there are likely some uses where consumers could benefit.

Reply:
After experimenting a bit, I think I'm going to lean toward just keeping `Pixel` as generic, and `bit_depth` as runtime. Adding another generic makes the interface more complicated, especially since consumers would have to support any potential bit depth they care about (8, 10, 12, possibly 16, possibly 14?). I think that unlike with integer type where there are heavy implications for optimization, most of what would involve bit depth can be handled with a bit shift, which will have to be done regardless of whether it's const or runtime.